frontend/src/app/map-view/comparison/comparison.component.ts

```diff
@@ -79,13 +87,17 @@ export class ComparisonComponent implements AfterViewInit {
     this.loadingCandidates = true;
 
     this.candidates$ = this.calcService.getMatchingCalculations(id.toString()).pipe(
-      withLatestFrom(this.translate.get('map.compare.choose-scenario')),
-      tap(([res, trans]) => {
+      tap((res) => {
+        const emptyList = res.filter(c => this.includeUnchanged || c.hasChanges).length === 0
         this.loadingCandidates = false;
-        this.bSelect.placeholder = trans;
-        this.bSelect.disabled = res.length === 0;
+        this.bSelect.placeholder =
+          this.translate.instant(
```

Comment:
I think this could be rewritten better, and I agree this belongs in the template. Honestly, I guess I'm (lazily) adapting a bit to some previous developers (slightly quirky) solution. So do I get away with it..🥺 or does it warrant a rewrite?